### PR TITLE
perf(cost): replace generic Data.contains with range(of:) in the rollout scan

### PR DESCRIPTION
**Pull request by VincentShipsIt**
## Summary
- Root cause of the never-ending "Updating…" spinner on Costs/Optimize: the Codex rollout scanner ran up to three `Data.contains(Data)` marker searches per line. Swift routes that through the generic `Collection.firstRange` path — benchmarked on the Mac Studio at **3.2 MB/s vs 1,787 MB/s** for `Data.range(of:)` (~560x). Over ~1 GB of rollouts each 15-second budget slice barely advanced, so `CostScanSession.isComplete` never turned true, `CostTracker.lastScanDate` never stamped (`cost-summary-v2.json` untouched since Aug 3), `needsMissingDailyUsageRefresh` stayed true, and the dashboard backfill re-triggered on every Costs/Optimize visit — endless spinner, one core pinned at 99% CPU.
- Fix: all four hot call sites now use `range(of:)`, with a comment pinning the constraint and a scoped `swiftlint:disable contains_over_range_nil_comparison`.
- Also fixes a #339 regression: the budgeted `makeScan` path silently dropped pricing provenance. It now records period/lifetime pricing from both providers, logs the pricing-table gap note, and carries `pricing` into the returned summary, mirroring `makeSummary`. A new `claudeProjectRoots` seam lets the regression test scan a fixture corpus instead of the real home directories.

## Verification
- `swift test --filter CostScanCollaboratorTests`: 40 passed (includes new `testBudgetedSummaryPreservesFullScanPricingProvenance`)
- `swift test --filter CostTrackerTests`: 56 passed
- Full `swift test` (Codex sandbox run): 1,881 executed, cost-scan suites green; only known sandbox-denied keychain/UI-agent/socket failures
- `swiftlint lint --strict --quiet`: clean
- Micro-benchmark (swiftc -O): `Data.contains` 108.6s vs `range(of:)` 0.20s for 200k searches over a 1.7 KB line

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved usage-event detection for more reliable cost scanning.
  * Ensured budgeted Claude summaries retain accurate pricing information.
  * Added diagnostics when pricing data is unavailable.

* **Improvements**
  * Claude scans can now use specified project locations or automatically discover relevant accounts.

### CODE REVIEW

File: MeterBar/Services/CostSummaryBuilder.swift [hunk 1]
```diff
@@ -42,12 +44,18 @@ enum CostSummaryBuilder {
...
+        claudeAccounts: [ClaudeCodeAccount],
+        claudeProjectRoots: [URL]?
     ) -> ScanWindows<CostScanResult> {
         var scan = ScanWindows(period: CostScanResult(), lifetime: CostScanResult(), cutoff: cutoffDate)
 
         if includeClaudeCode {
-            let claude = ClaudeCostScanner.scanSessions(since: cutoffDate, claudeAccounts: claudeAccounts)
+            let claude: ScanWindows<ClaudeSessionTotals>
+            if let claudeProjectRoots {
+                claude = ClaudeCostScanner.scanSessions(since: cutoffDate, projectRoots: claudeProjectRoots)
```
[the remark this answers is not in the document]
Comment (VincentShipsIt) [reply MeterBar/Services/CostSummaryBuilder.swift:55]: Valid, and tracked as #359 rather than fixed here — this PR only swaps `Data.contains` for `range(of:)` in the rollout scan, and the gap it points at is older than the diff.

Two things bound the risk today. `projectRoots(accounts:)` already dedups *exact* paths (`seen.insert(standardized.path)`), and the roots it yields in production are sibling `<config>/projects` directories, never nested — so an overlapping-but-not-identical pair needs either a `CLAUDE_CONFIG_DIR` naming an ancestor of another root, or the `claudeProjectRoots` seam this PR added for tests (production still passes nil and routes through `claudeAccounts`).

The real fix is to carry the same `file.cacheKey` set across the root loop that `scanRoots` already uses, so both scan paths agree on what one transcript is. That is #359, with a test asserting overlapping roots and a single root produce identical totals on both paths.
